src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -24,30 +27,37 @@ object Lorre extends App with LazyLogging {
       | Please provide a valid network as an argument to the command line""".stripMargin)
       sys.exit(1)
     }
-  
+
 
   private val conf = ConfigFactory.load
   private val awaitTimeInSeconds = conf.getInt("dbAwaitTimeInSeconds")
   private val sleepIntervalInSeconds = conf.getInt("lorre.sleepIntervalInSeconds")
   private val feeUpdateInterval = conf.getInt("lorre.feeUpdateInterval")
   private val purgeAccountsInterval = conf.getInt("lorre.purgeAccountsInterval")
 
+  //re-use the dispatcher pool from the node interface
+  implicit val dispatcher = TezosNodeInterface.system.dispatcher
+
   lazy val db = DatabaseUtil.db
   val tezosNodeOperator = new TezosNodeOperator(TezosNodeInterface)
 
   @tailrec
   def mainLoop(iteration: Int): Unit = {
-      processTezosBlocks()
-      processTezosAccounts()
-      if (iteration % feeUpdateInterval == 0) {
-        FeeOperations.processTezosAverageFees()
-      }
-      if (iteration % purgeAccountsInterval == 0) {
-        TezosDatabaseOperations.purgeOldAccounts()
-      }
-      logger.info("Taking a nap")
-      Thread.sleep(sleepIntervalInSeconds * 1000)
-      mainLoop(iteration + 1)
+    Await.ready(
+      processTezosBlocks().flatMap(
+        _ =>
+          processTezosAccounts()
+      )
+    , atMost = Duration.Inf)
```

Comment:
what purpose does this serve? Speedup?

Reply:
As the operations return `Future` values now, the call to the `processXXX` methods will immediately return without waiting for the answer.

We need to sequence those operations instead (monadically in this case) because we can't start processing accounts before all the blocks are stored.
Additionally the same is true for fees and purge operations, they need to start only after the previous have finished.

In this case I didn't chain them monadically, but blocked on the results, because eventually the code will recursively call itself, and we can't do this within a different thread on a callback if we want it to be tail-recursive (preventing stack-overflows when running for many cycles)